src/Microsoft.Azure.Batch.SoftwareEntitlement/TokenGenerator.cs

```diff
@@ -114,6 +114,33 @@ private List<Claim> CreateClaims(NodeEntitlements entitlements)
                 claims.Add(new Claim(Claims.VirtualMachineId, entitlements.VirtualMachineId));
             }
 
+            _logger.LogDebug("CPU core count: {CpuCoreCount}", entitlements.CpuCoreCount);
+            claims.Add(new Claim(Claims.CpuCoreCount, entitlements.CpuCoreCount.ToString()));
+
+            if (!string.IsNullOrEmpty(entitlements.BatchAccountId))
+            {
+                _logger.LogDebug("Batch Account Id: {BatchAccountId}", entitlements.BatchAccountId);
+                claims.Add(new Claim(Claims.BatchAccountId, entitlements.BatchAccountId));
+            }
+
+            if (!string.IsNullOrEmpty(entitlements.PoolId))
+            {
+                _logger.LogDebug("Pool Id: {PoolId}", entitlements.PoolId);
+                claims.Add(new Claim(Claims.PoolId, entitlements.PoolId));
+            }
+
+            if (!string.IsNullOrEmpty(entitlements.JobId))
+            {
+                _logger.LogDebug("Job Id: {JobId}", entitlements.JobId);
+                claims.Add(new Claim(Claims.JobId, entitlements.JobId));
+            }
```

Comment:
Perhaps this could use the same helper method I suggested for `NodeEntitlementReader`.